description: fix warnings in embedded libs
author: Michael Gilbert <mgilbert@debian.org>

--- a/libs/xml2/Makefile.in
+++ b/libs/xml2/Makefile.in
@@ -1,6 +1,6 @@
 EXTLIB    = libxml2.a
 EXTRAINCL = $(XML2_PE_CFLAGS)
-EXTRADEFS = -DHAVE_VA_COPY
+EXTRADEFS = -DHAVE_VA_COPY -Wno-format-extra-args
 
 SOURCES = \
 	HTMLparser.c \
--- a/libs/faudio/src/FAudio_internal.h
+++ b/libs/faudio/src/FAudio_internal.h
@@ -87,8 +87,8 @@
 #define FAudio_snprintf snprintf
 #define FAudio_vsnprintf vsnprintf
 #define FAudio_getenv getenv
-#define FAudio_PRIu64 PRIu64
-#define FAudio_PRIx64 PRIx64
+#define FAudio_PRIu64 "I64u"
+#define FAudio_PRIx64 "I64x"
 
 extern void FAudio_Log(char const *msg);
 
--- a/libs/mpg123/src/libmpg123/gapless.h
+++ b/libs/mpg123/src/libmpg123/gapless.h
@@ -81,7 +81,7 @@ static void frame_buffercheck(mpg123_han
 		}
 		if(VERBOSE3)
 			fprintf(stderr, "\nNote: Cut frame %" PRIi64 " buffer on end of stream to %"
-				 PRIi64 " samples, fill now %zu bytes.\n"
+				 PRIi64 " samples, fill now %u bytes.\n"
 			,	fr->num, (fr->num == fr->lastframe ? fr->lastoff : 0), fr->buffer.fill);
 	}
 
@@ -106,7 +106,7 @@ static void frame_buffercheck(mpg123_han
 
 		if(VERBOSE3)
 			fprintf(stderr, "\nNote: Cut frame %" PRIi64
-				" buffer on beginning of stream by %" PRIi64 " samples, fill now %zu bytes.\n"
+				" buffer on beginning of stream by %" PRIi64 " samples, fill now %u bytes.\n"
 			,	fr->num, fr->firstoff, fr->buffer.fill);
 		/* We can only reach this frame again by seeking. And on seeking, firstoff will be recomputed.
 		   So it is safe to null it here (and it makes the if() decision abort earlier). */
--- a/libs/mpg123/src/libmpg123/readers.c
+++ b/libs/mpg123/src/libmpg123/readers.c
@@ -42,7 +42,7 @@ static ptrdiff_t fdread(mpg123_handle *f
 		int ret = fr->rdat.r_read64(fr->rdat.iohandle, buf, count, &got);
 		if(ret<0)
 		{
-			if(NOQUIET) merror("error reading %zu bytes", count);
+			if(NOQUIET) merror("error reading %u bytes", count);
 			return -1;
 		}
 		if(VERBOSE3) mdebug("read %zu bytes of %zu", got, count);
--- a/libs/mpg123/src/libmpg123/frame.c
+++ b/libs/mpg123/src/libmpg123/frame.c
@@ -193,7 +193,7 @@ int INT123_frame_outbuffer(mpg123_handle
 		{
 			fr->err = MPG123_BAD_BUFFER;
 			if(NOQUIET)
-				merror("have external buffer of size %zu, need %zu", fr->buffer.size, size);
+				merror("have external buffer of size %u, need %u", fr->buffer.size, size);
 			return MPG123_ERR;
 		}
 	}
--- a/libs/mpg123/src/libmpg123/id3.c
+++ b/libs/mpg123/src/libmpg123/id3.c
@@ -567,7 +567,7 @@ static void process_comment(mpg123_handl
 
 	if(realsize < (size_t)(descr-realdata))
 	{
-		if(NOQUIET) error1("Invalid frame size of %zu (too small for anything).", realsize);
+		if(NOQUIET) error1("Invalid frame size of %u (too small for anything).", realsize);
 		return;
 	}
 	if(encoding > mpg123_id3_enc_max)
@@ -622,8 +622,8 @@ static void process_comment(mpg123_handl
 
 	if(VERBOSE4) /* Do _not_ print the verbatim text: The encoding might be funny! */
 	{
-		fprintf(stderr, "Note: ID3 comm/uslt desc of length %zu.\n", xcom->description.fill);
-		fprintf(stderr, "Note: ID3 comm/uslt text of length %zu.\n", xcom->text.fill);
+		fprintf(stderr, "Note: ID3 comm/uslt desc of length %u.\n", xcom->description.fill);
+		fprintf(stderr, "Note: ID3 comm/uslt text of length %u.\n", xcom->text.fill);
 	}
 	/* Look out for RVA info only when we really deal with a straight comment. */
 	if(tt == comment && localcom.description.fill > 0)
--- a/libs/mpg123/src/libmpg123/libmpg123.c
+++ b/libs/mpg123/src/libmpg123/libmpg123.c
@@ -923,7 +923,7 @@ static void decode_the_frame(mpg123_hand
 		if(fr->buffer.fill < needed_bytes)
 		{
 			if(VERBOSE2)
-				fprintf( stderr, "Note: broken frame %li, filling up with %zu zeroes, from %zu\n"
+				fprintf( stderr, "Note: broken frame %li, filling up with %u zeroes, from %u\n"
 				,	(long)fr->num, (needed_bytes-fr->buffer.fill), fr->buffer.fill );
 
 			/*
