description: regenerate vulkan files
forwarded: not-needed
author: Michael Gilbert <mgilbert@debian.org>

--- a/dlls/winevulkan/make_vulkan
+++ b/dlls/winevulkan/make_vulkan
@@ -61,8 +61,18 @@ from enum import Enum
 #   Array length is often stored in a previous parameter or another structure
 #   member and thus needs careful parsing.
 
+class MakeVulkanError(Exception):
+    def __init__(self, message):
+        super().__init__(message)
+
+class MakeVulkanHandler(logging.StreamHandler):
+    def emit(self, record):
+        if record.levelno in (logging.ERROR, logging.CRITICAL):
+            raise MakeVulkanError(record.getMessage())
+        super().emit(record)
+
 LOGGER = logging.Logger("vulkan")
-LOGGER.addHandler(logging.StreamHandler())
+LOGGER.addHandler(MakeVulkanHandler())
 
 VK_XML_VERSION = "1.4.303"
 WINE_VK_VERSION = (1, 4)
@@ -2700,7 +2710,12 @@ class VkGenerator(object):
 
     def _generate_copyright(self, f, spec_file=False):
         f.write("# " if spec_file else "/* ")
-        f.write("Automatically generated from Vulkan vk.xml and video.xml; DO NOT EDIT!\n")
+        f.write("Automatically generated from ")
+        f.write(self.registry._vk)
+        f.write(" and ")
+        f.write(self.registry._video)
+        f.write("; DO NOT EDIT!\n")
+
         lines = ["", "This file is generated from Vulkan vk.xml file covered",
             "by the following copyright and permission notice:"]
         lines.extend([l.rstrip(" ") for l in self.registry.copyright.splitlines()])
@@ -3198,6 +3213,8 @@ class VkRegistry(object):
         self.funcpointers = None
         self.handles = None
         self.structs = None
+        self._vk = vk_xml
+        self._video = video_xml
 
         # We aggregate all types in here for cross-referencing.
         self.funcs = {}
@@ -3844,11 +3861,6 @@ def set_working_directory():
     path = os.path.dirname(path)
     os.chdir(path)
 
-def download_vk_xml(dst_filename, src_filename):
-    url = "https://raw.githubusercontent.com/KhronosGroup/Vulkan-Docs/v{0}/xml/{1}".format(VK_XML_VERSION, src_filename)
-    if not os.path.isfile(dst_filename):
-        urllib.request.urlretrieve(url, dst_filename)
-
 def main():
     parser = argparse.ArgumentParser()
     parser.add_argument("-v", "--verbose", action="count", default=0, help="increase output verbosity")
@@ -3865,23 +3877,15 @@ def main():
 
     set_working_directory()
 
-    if "XDG_CACHE_HOME" in os.environ:
-        cache = os.path.expandvars("$XDG_CACHE_HOME/wine")
-    else:
-        cache = os.path.expandvars("$HOME/.cache/wine")
-    os.makedirs(cache, exist_ok=True)
-
     if args.xml:
         vk_xml = args.xml
     else:
         vk_xml = os.path.join(cache, "vk-{0}.xml".format(VK_XML_VERSION))
-        download_vk_xml(vk_xml, "vk.xml")
 
     if args.video_xml:
         video_xml = args.video_xml
     else:
         video_xml = os.path.join(cache, "video-{0}.xml".format(VK_XML_VERSION))
-        download_vk_xml(video_xml, "video.xml")
 
     registry = VkRegistry(vk_xml, video_xml)
     generator = VkGenerator(registry)
