description: reflect GPL permisiveness in embedding characteristics of fonts
author: Michael Gilbert <mgilbert@debian.org>

--- a/fonts/courier.sfd
+++ b/fonts/courier.sfd
@@ -15,7 +15,7 @@ InvalidEm: 0
 LayerCount: 2
 Layer: 0 0 "Back" 1
 Layer: 1 0 "Fore" 0
-FSType: 8
+FSType: 0
 OS2Version: 0
 OS2_WeightWidthSlopeOnly: 0
 OS2_UseTypoMetrics: 0
--- a/fonts/fixedsys.sfd
+++ b/fonts/fixedsys.sfd
@@ -15,7 +15,7 @@ InvalidEm: 0
 LayerCount: 2
 Layer: 0 0 "Back" 1
 Layer: 1 0 "Fore" 0
-FSType: 8
+FSType: 0
 OS2Version: 1
 OS2_WeightWidthSlopeOnly: 0
 OS2_UseTypoMetrics: 0
--- a/fonts/fixedsys_jp.sfd
+++ b/fonts/fixedsys_jp.sfd
@@ -16,7 +16,7 @@ LayerCount: 2
 Layer: 0 0 "Back" 1
 Layer: 1 0 "Fore" 0
 XUID: [1021 516 2032022868 7388451]
-FSType: 8
+FSType: 0
 OS2Version: 1
 OS2_WeightWidthSlopeOnly: 0
 OS2_UseTypoMetrics: 1
--- a/fonts/ms_sans_serif.sfd
+++ b/fonts/ms_sans_serif.sfd
@@ -15,7 +15,7 @@ InvalidEm: 0
 LayerCount: 2
 Layer: 0 0 "Back" 1
 Layer: 1 0 "Fore" 0
-FSType: 8
+FSType: 0
 OS2Version: 0
 OS2_WeightWidthSlopeOnly: 0
 OS2_UseTypoMetrics: 0
--- a/fonts/small_fonts.sfd
+++ b/fonts/small_fonts.sfd
@@ -16,7 +16,7 @@ LayerCount: 2
 Layer: 0 0 "Back" 1
 Layer: 1 0 "Fore" 0
 XUID: [1021 180 78732788 213171]
-FSType: 8
+FSType: 0
 OS2Version: 0
 OS2_WeightWidthSlopeOnly: 0
 OS2_UseTypoMetrics: 0
--- a/fonts/small_fonts_jp.sfd
+++ b/fonts/small_fonts_jp.sfd
@@ -16,7 +16,7 @@ LayerCount: 2
 Layer: 0 0 "Back" 1
 Layer: 1 0 "Fore" 0
 XUID: [1021 180 78732788 213171]
-FSType: 8
+FSType: 0
 OS2Version: 0
 OS2_WeightWidthSlopeOnly: 0
 OS2_UseTypoMetrics: 0
--- a/fonts/system.sfd
+++ b/fonts/system.sfd
@@ -15,7 +15,7 @@ InvalidEm: 0
 LayerCount: 2
 Layer: 0 0 "Back" 1
 Layer: 1 0 "Fore" 0
-FSType: 8
+FSType: 0
 OS2Version: 0
 OS2_WeightWidthSlopeOnly: 0
 OS2_UseTypoMetrics: 0
