description: fix incorrect handling of WRITECOPY memory protection
author: Fabian Maurer
bug: https://bugs.winehq.org/show_bug.cgi?id=29384#c56
bug: https://bugs.winehq.org/show_bug.cgi?id=56711#c9

--- a/dlls/kernelbase/memory.c
+++ b/dlls/kernelbase/memory.c
@@ -547,7 +547,9 @@ BOOL WINAPI DECLSPEC_HOTPATCH  VirtualLo
  */
 BOOL WINAPI DECLSPEC_HOTPATCH VirtualProtect( void *addr, SIZE_T size, DWORD new_prot, DWORD *old_prot )
 {
-    return VirtualProtectEx( GetCurrentProcess(), addr, size, new_prot, old_prot );
+    BOOL ret = VirtualProtectEx( GetCurrentProcess(), addr, size, new_prot, old_prot );
+    if (old_prot && *old_prot == PAGE_WRITECOPY) *old_prot = PAGE_READWRITE;
+    return ret;
 }
 
 
